src/audio-manager.ts

```diff
@@ -1,261 +1,533 @@
+    Music,
+    /** Connects directly to output. */
+    Master,
+}
+
+/**
+ * Enumerates the possible states of playback for audio sources.
+ */
+export enum PlayState {
+    /** The source has loaded and is ready to be played */
+    Ready,
+    /** The source has started playing */
+    Playing,
+    /** The source has stopped */
+    Stopped,
+}
+
+/**
+ * Represents a combination of a unique identifier and a play state.
+ */
+type PlayStateWithID = {
+    /** Unique identifier associated with the audio source. */
+    id: number;
+    /** Current state of playback for the audio source. */
+    state: PlayState;
+};
 
 /**
- * AudioManager loads and manages audio files from which PlayableNodes are created.
+ * Combines all settings for configuring playback in the AudioManager.
+ *
+ * @note The playOneShot() function utilizes this configuration for consistent playback settings. However, one-shots
+ * can only change their volume and position.
+ */
+export type PlayConfig = {
+    /** Sets the volume of the player (0-1) */
+    volume?: number;
+    /** Whether to loop the audio */
+    loop?: boolean;
+    /**
+     * Sets the position of the audio source and makes it spatial.
+     *
+     * @note Panned audio will always use HRTF for spatialization.
+     * @warning For this to work correctly, the audio-listener needs to be set up!
+     */
+    position?: Float32Array;
+    /** Sets the channel on which the audio will be played. */
+    channel?: AudioChannel;
+    /**
+     * Whether the audio has priority or not. If not, playback can be stopped to free up a player when no others are
+     * available.
+     */
+    priority?: boolean;
+};
+
+/**
+ * Default number of one-shot players.
+ */
+export const DEF_ONESHOT_PLAYER_COUNT = 16;
+
+/**
+ * Default number of regular players.
+ */
+export const DEF_PLAYER_COUNT = 16;
+
+const SHIFT_AMOUNT = 16;
+const MAX_NUMBER_OF_INSTANCES = (1 << SHIFT_AMOUNT) - 1;
+
+/**
+ * Manages audio files and players, providing control over playback on three audio channels.
+ *
+ * @classdesc
+ * The AudioManager handles audio files and players, offering control over playback on three distinct channels.
+ * It supports two types of players: OneShot players, which play audio once and return, and regular players.
+ * OneShot players are less configurable but more performant than regular players.
+ * @see AudioChannel
+ *
+ * @note The AudioManager is able to play audio with spatial positioning. Keep in mind that for this to work
+ * correctly, you will need to set up the `audio-listener` component!
  *
  * @example
  * ```js
- * async start() {
- *      this.audio = await audioManager.load('click.wav');
+ * enum Sounds {
+ *      Click,
+ *      GunShot,
  * }
+ *
+ * // AudioManager can't be constructed in a non-browser environment!
+ * export const am = window.AudioContext ? new AudioManager() : null;
+ *
+ * am.load('path/to/click.wav', Sounds.Click);
+ * am.load('path/to/gunshot.wav', Sounds.GunShot);
```

Comment:
This code will throw, since `am` might be null :)

I think you'll need to have a `if(am !== null)` check here.